### PR TITLE
[MNG-7193] Introduce MAVEN_ARGS environment variable

### PR DESCRIPTION
**Pull request by michael-o**
Add args directly after the class name and leave users the option to override
with maven.config or CLI args.

This closes #637

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Review by rfscholte — APPROVED** [review]
This needs to be documented somewhere as well.

**Comment by michael-o** [comment]
> This needs to be documented somewhere as well.

You are right. I will locate the appropriate document before merging this.

**Review by mthmulders — APPROVED** [review]
Did the [same test as three weeks ago](https://github.com/apache/maven/pull/602#issuecomment-981747872) but this time it behaves as expected 👍🏻

**Comment by mthmulders** [comment]
I guess this means we can close #602?

**Comment by michael-o** [comment]
> 

I will supersede #602 with two alternative PRs.

**Comment by michael-o** [comment]
@rfscholte Updated inline scripts and add https://github.com/apache/maven-site/pull/280.

File: apache-maven/src/assembly/shared/validate.cmd [hunk 1]
```diff
@@ -24,6 +24,7 @@
 @REM   MAVEN_BATCH_ECHO  (Optional) Set to 'on' to enable the echoing of the batch commands.
 @REM   MAVEN_BATCH_PAUSE (Optional) set to 'on' to wait for a key stroke before ending.
 @REM   MAVEN_OPTS        (Optional) Java runtime options used when Maven is executed.
+@REM   MAVEN_ARGS        (Optional) Arguments passed to Maven before CLI arguments.
```
Comment (rfscholte) [inline apache-maven/src/assembly/shared/validate.cmd:27]: Assuming these are in alphabetic order, this is not the right location

Comment (michael-o) [reply apache-maven/src/assembly/shared/validate.cmd:27]: While you may be right I think they sorted logically by usage. I make it alphabetically, no issue.

**Comment by rfscholte** [comment]
Must also end up in https://github.com/apache/maven-wrapper

**Comment by michael-o** [comment]
@rfscholte While I agree those scripts are completely Maven 3. I.e, they would require to migrate to Maven 4 approach first. (reduced set). Cherry-picking pieces does not make sense. I can create a ticket to track this.

**Pull request closed without merging**